src/dotnet/SlnProjectExtensions.cs

```diff
@@ -12,17 +12,29 @@ public static class SlnProjectExtensions
     {
         public static IList<string> GetSolutionFoldersFromProject(this SlnProject project)
         {
-            var currentDirString = $".{Path.DirectorySeparatorChar}";
+            var solutionFolders = new List<string>();
 
-            var directoryPath = Path.GetDirectoryName(project.FilePath);
-            if (directoryPath.StartsWith(currentDirString))
+            var projectFilePath = project.FilePath;
+            if (!projectFilePath.StartsWith(".."))
             {
-                directoryPath = directoryPath.Substring(currentDirString.Length);
+                var currentDirString = $".{Path.DirectorySeparatorChar}";
+                if (projectFilePath.StartsWith(currentDirString))
+                {
+                    projectFilePath = projectFilePath.Substring(currentDirString.Length);
+                }
+
+                var projectDirectoryPath = Path.GetDirectoryName(projectFilePath);
+                if (!string.IsNullOrEmpty(projectDirectoryPath))
```

Comment:
Could you suggest some good names? We need to GetDirectoryName twice because the first one gets the parent directory for the proj file, but the project directory should not be turned into a solution folder (see the associated issue for more context). So that's why I need to call it again...